### PR TITLE
985 client side validation

### PR DESCRIPTION
**Pull request by norisalsaadie**
Closes https://github.com/libero/reviewer/issues/985

### CODE REVIEW

File: tests/page-objects/EditorPage.ts [hunk 1]
```diff
@@ -37,24 +37,43 @@ export class EditorPage {
...
+        await t.click(firstAddButton);
+        const secondAddButton = Selector('.people-picker__modal_list--item:nth-child(even) .pod__button');
+        await t.expect(secondAddButton.visible).ok();
+        await t.click(secondAddButton);
+        await t.expect(Selector('.people-picker__selected-tabs').child('.people-picker__selected-tab').count).eql(2);
+        await t.click(Selector('.modal__buttons_container').find('.button--primary'));
+        await t.expect(Selector('.modal .modal__fullscreen').exists).eql(false);
+        await t.expect(picker.find('.selected_people_list__item').count).eql(3);
+    }
+
```
Comment (will-byrne) [inline tests/page-objects/EditorPage.ts:58]: Can this not be abstracted as its a clone of the addPersonToPeoplePicker function.

File: src/initial-submission/components/EditorsForm.tsx [hunk 2]
```diff
@@ -128,10 +174,12 @@ const EditorsForm = ({ initialValues, ButtonComponent }: StepProps): JSX.Element
         },
         mode: 'onBlur',
         validateCriteriaMode: 'all',
-        validationSchema: schema,
+        validationResolver,
     });
     const [saveCallback] = useMutation<Submission>(saveEditorsPageMutation);
 
+    console.log('errors', errors);
```
Comment (Cooryd) [inline src/initial-submission/components/EditorsForm.tsx:181]: console.log

File: tests/page-objects/EditorPage.ts (hunk 1 above)
Comment (norisalsaadie) [reply tests/page-objects/EditorPage.ts]: Addressed

File: tests/page-objects/EditorPage.ts [hunk 3]
```diff
@@ -43,24 +43,34 @@ export class EditorPage {
         await this.addOpposingReviewer();
     }
 
-    private async addPersonToPeoplePicker(picker: Selector): Promise<void> {
+    private async addPersonToPeoplePicker(picker: Selector, number = 1): Promise<void> {
         await t.expect(picker.find('.selected_people_list__item').count).eql(1);
         const addButton = picker.find('.pod__button');
         await t.click(addButton);
+        await this.assertPeoplePickerSearch();
```
Comment (will-byrne) [inline tests/page-objects/EditorPage.ts:50]: merge issue, this should be removed again.

File: tests/page-objects/EditorPage.ts [hunk 4]
```diff
@@ -43,24 +43,34 @@ export class EditorPage {
...
+
+        const buttons = [];
+        for (let i = 0; i < number; i++) {
+            buttons.push(addButtonSelector.nth(i));
+        }
+
+        for await (const button of buttons) {
+            await t.expect(button.visible).ok();
+            await t.click(button);
+        }
```
Comment (will-byrne) [inline tests/page-objects/EditorPage.ts:64]: could this be made into a single loop?
something like 
```ts
for (let i=0; i < number; i++) {
    const button = addButtonSelector.nth(i);
    await t.expect(button.visible).ok();
    await t.click(button);
}
```

File: package.json [hunk 5]
```diff
@@ -43,7 +43,8 @@
         "@types/apollo-upload-client": "^8.1.3",
         "@types/graphql": "^14.2.3",
         "@types/jest": "^24.0.15",
-        "@types/lodash": "^4.14.138",
+        "@types/lodash": "^4.14.154",
+        "@types/nightwatch": "^1.1.3",
```
Comment (will-byrne) [inline package.json:47]: nightwatch should probably not be re-added 😄

Comment (norisalsaadie) [reply package.json]: Done. It gets auto install for whatever reason

File: tests/page-objects/EditorPage.ts (hunk 4 above)
Comment (norisalsaadie) [reply tests/page-objects/EditorPage.ts]: done

File: tests/page-objects/EditorPage.ts (hunk 3 above)
Comment (norisalsaadie) [reply tests/page-objects/EditorPage.ts]: done

**Pull request merged**